shell/platform/android/io/flutter/plugin/editing/ListenableEditingState.java

```diff
@@ -240,6 +271,23 @@ public final int getComposingEnd() {
     return BaseInputConnection.getComposingSpanEnd(this);
   }
 
+  @Override
+  public void setSpan(Object what, int start, int end, int flags) {
+    super.setSpan(what, start, end, flags);
+    // Setting a span does not involve mutating the text value in the editing state. Here we create
+    // a non text update delta with any updated selection and composing regions.
+    mBatchTextEditingDeltas.add(
+        new TextEditingDelta(
```

Comment:
We should make a "nonTextUpdate" delta be its own constructor. It is not always obvious to pass `"", -1, -1`. A separate constructor for a separate case would prevent any accidental misuse or misconfiguration of this particular type of delta.

Reply:
This makes sense, I do this already in the iOS PR.